packages/python-editor/src/KernelDropdown.tsx

```diff
@@ -16,128 +16,86 @@
 
 import { ReactWidget } from '@jupyterlab/apputils';
 import { KernelSpec } from '@jupyterlab/services';
-import { HTMLSelect } from '@jupyterlab/ui-components';
-import React from 'react';
 
-import { PythonRunner } from './PythonRunner';
+import React, {
+  forwardRef,
+  useImperativeHandle,
+  useState,
+  RefObject
+} from 'react';
 
-const DROPDOWN_CLASS = 'jp-Notebook-toolbarCellTypeDropdown bp3-minimal';
+const KERNEL_SELECT_CLASS = 'elyra-ScriptEditor-KernelSelector';
 
-/**
- * Class: Holds properties for toolbar dropdown.
- */
-class DropDownProps {
-  runner: PythonRunner;
-  updateKernel: Function;
+export interface ISelect {
+  getSelection: () => string;
 }
 
-/**
- * Class: Holds kernel state property.
- */
-class DropDownState {
-  kernelSpecs: KernelSpec.ISpecModels;
+class Props {
+  specs: KernelSpec.ISpecModels;
 }
 
 /**
- * Class: A toolbar dropdown component populated with available kernel specs.
+ * A toolbar dropdown component populated with available kernel specs.
  */
-class DropDown extends React.Component<DropDownProps, DropDownState> {
-  private updateKernel: Function;
-  private kernelOptionElems: Record<string, any>[];
-
-  /**
-   * Construct a new dropdown widget.
-   */
-  constructor(props: DropDownProps) {
-    super(props);
-    this.state = { kernelSpecs: null };
-    this.updateKernel = this.props.updateKernel;
-    this.kernelOptionElems = [];
-    this.getKernelSPecs();
-  }
-
-  /**
-   * Function: Gets kernel specs and state.
-   */
-  private async getKernelSPecs(): Promise<void> {
-    const specs: KernelSpec.ISpecModels = await this.props.runner.getKernelSpecs();
-    this.filterPythonKernels(specs);
-
-    // Set kernel to default
-    this.updateKernel(specs.default);
-
-    this.createOptionElems(specs);
-    this.setState({ kernelSpecs: specs });
+// eslint-disable-next-line react/display-name
```

Comment:
afaik you shouldn't need this since you're using an arrow function

Reply:
deleting that line I get this error:
```
  40:18  error  Component definition is missing display name  react/display-name

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
```

Reply:
hmm that's not the error I thought you were getting, seems like a react thing, @bourdakos1 what is the correct way to write this so we don't have to disable the linter?

Reply:
Umm I think its because we are passing an anonymous function to forwardRef. I think it’s fine to ignore this, but you could also try setting the component to a variable and pass that to forwardRef and if it’s still complaining you can do .displayName = xxx